fuzzing/nxt_basic_fuzz.c

```diff
@@ -81,11 +159,78 @@ nxt_time_fuzz(const u_char *data, size_t size)
 }
 
 
+void
+nxt_uri_fuzz(const u_char *data, size_t size)
+{
+    u_char  *dst;
+
+    dst = nxt_zalloc(size * 3);
+    if (dst == NULL) {
+        return;
+    }
+
+    nxt_decode_uri(dst, (u_char *) data, size);
+    nxt_decode_uri_plus(dst, (u_char *) data, size);
+
+    nxt_encode_uri(NULL, (u_char *) data, size);
+    nxt_encode_uri(dst, (u_char *) data, size);
+
+    nxt_free(dst);
+}
+
+
 void
 nxt_utf8_fuzz(const u_char *data, size_t size)
 {
     const u_char  *in;
 
     in = data;
     nxt_utf8_decode(&in, data + size);
+
+    nxt_utf8_casecmp((u_char *) "ABC АБВ ΑΒΓ",
```

Comment:
That cast isn't needed is it? The prototype takes a `const u_char *`...

Reply:
I will re-check this was from the test, BTW.

Reply:
Yeah, I noticed that, I think it's erroneous.

Reply:
Well, necessary evil, I guess.

```c
  CC     build/fuzzing/nxt_basic_fuzz.o
fuzzing/nxt_basic_fuzz.c:190:22: error: passing 'const char[18]' to parameter of type 'const u_char *' (aka 'const unsigned char *') converts between pointers to integer types where one is of the unique plain 'char' type and the other is not [-Werror,-Wpointer-sign]
  190 |     nxt_utf8_casecmp("ABC АБВ ΑΒΓ",
      |                      ^~~~~~~~~~~~~
src/nxt_utf8.h:22:53: note: passing argument to parameter 'start1' here
   22 | NXT_EXPORT nxt_int_t nxt_utf8_casecmp(const u_char *start1,
      |                                                     ^
1 error generated.
make: *** [build/Makefile:1858: build/fuzzing/nxt_basic_fuzz.o] Error 1
```

Reply:
Ah, yes. I do have plans to eventually get rid (mostly) of our use of `u_char`. But for now that should of course be `(const u_char *)`

Reply:
Done: https://github.com/nginx/unit/pull/1372/files#diff-568f7f9c4b9b4ba9069a59ffd500a9bf3c1404664b0c7ca4146de69dabd7e558R190